src/adapter/properties.rs

```diff
@@ -312,20 +312,20 @@ pub(super) fn resolve_method_receiver_property<'a, V: AsVertex<Vertex<'a>> + 'a>
 ) -> ContextOutcomeIterator<'a, V, FieldValue> {
     match property_name {
         "by_value" => resolve_property_with(contexts, |vertex| {
-            let receiver = vertex.as_method_receiver().expect("not a MethodReceiver");
+            let receiver = vertex.as_receiver().expect("not a MethodReceiver");
             receiver.by_value().into()
         }),
         "by_reference" => resolve_property_with(contexts, |vertex| {
-            let receiver = vertex.as_method_receiver().expect("not a MethodReceiver");
+            let receiver = vertex.as_receiver().expect("not a MethodReceiver");
             receiver.by_reference().into()
         }),
         "by_mut_reference" => resolve_property_with(contexts, |vertex| {
-            let receiver = vertex.as_method_receiver().expect("not a MethodReceiver");
+            let receiver = vertex.as_receiver().expect("not a MethodReceiver");
             receiver.by_mut_reference().into()
         }),
         "kind" => resolve_property_with(contexts, |vertex| {
-            let receiver = vertex.as_method_receiver().expect("not a MethodReceiver");
-            receiver.kind().into()
+            let receiver = vertex.as_receiver().expect("not a MethodReceiver");
+            receiver.kind().as_ref().into()
```

Comment:
Almost certainly already on your radar, but just in case: we want all the `.expect()` messages to reference `Receiver` instead of `MethodReceiver`.